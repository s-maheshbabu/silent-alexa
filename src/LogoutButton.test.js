import React from "react";
import { shallow, mount } from "enzyme";
import LogoutButton from "./LogoutButton";

it("renders correctly (snapshot testing)", () => {
  const wrapper = shallow(<LogoutButton />);
  expect(wrapper).toMatchSnapshot();

  wrapper.unmount();
});

it("verifies that the callback function is called when logout button is clicked", () => {
  const mockCallBack = jest.fn();
  const button = shallow(<LogoutButton onClick={mockCallBack} />).find(
    "button"
  );
  expect(button.length).toBe(1);
  button.simulate("click");

  // Verify mockCallBack called once
  expect(mockCallBack.mock.calls.length).toBe(1);
});
