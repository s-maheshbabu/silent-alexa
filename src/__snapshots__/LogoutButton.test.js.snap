// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly (snapshot testing) 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Logout
</button>
`;
